src/DblParser/Desugar.ml

```diff
@@ -816,4 +837,45 @@ and make_handle ~pos lbl_opt eff_opt pat body hcs =
       fin_clauses = fcs
     })
 
-let tr_program (p : Raw.program) = { p with data = tr_defs p.data }
+and generate_accessor_method_pattern named_type_args type_name =
+  let make data = { data; pos=Position.nowhere } in
+  let create_mapping i arg =
+    let old_name, new_name =
+      match (snd arg.data).data with
+      | TA_Var(name, _) -> name, name ^ "#TA_Var#" ^ string_of_int i
+      | TA_Effect -> "", "#TA_Effect#" ^ string_of_int i
+      | TA_Wildcard -> "", "#TA_Wildcard#" ^ string_of_int i
+    in
+    make (TNVar old_name, make (TA_Var(new_name, make (KWildcard)))), new_name
+  in
+  let new_named_type_args, new_names = List.split (List.mapi create_mapping named_type_args) in
```

Comment:
This line doesn't fit in 80 columns. Therefore, I cannot browse this code on [VT100](https://en.wikipedia.org/wiki/VT100) terminal :smile:.